src/category_theory/limits/preserves/shapes.lean

```diff
@@ -14,32 +14,203 @@ open category_theory category_theory.category category_theory.limits
+
+variables [has_product (λ (j : J), G.obj (f j))]
+
+/--
+If `G` preserves limits, we have an isomorphism from the image of a product to the product of the
+images.
 -/
 -- TODO perhaps weaken the assumptions here, to just require the relevant limits?
-def preserves_products_iso {J : Type v} (f : J → C) [has_limits C] [has_limits D] :
-  G.obj (pi_obj f) ≅ pi_obj (λ j, G.obj (f j)) :=
-preserves_limit_iso G (discrete.functor f) ≪≫
-  has_limit.iso_of_nat_iso (discrete.nat_iso (λ j, iso.refl _))
+def preserves_products_iso : G.obj (∏ f) ≅ ∏ (λ j, G.obj (f j)) :=
+is_limit.cone_point_unique_up_to_iso (preserves_the_product G f) (limit.is_limit _)
 
 @[simp, reassoc]
-lemma preserves_products_iso_hom_π
-  {J : Type v} (f : J → C) [has_limits C] [has_limits D] (j) :
+lemma preserves_products_iso_hom_π (j) :
   (preserves_products_iso G f).hom ≫ pi.π _ j = G.map (pi.π f j) :=
-by simp [preserves_products_iso]
+is_limit.cone_point_unique_up_to_iso_hom_comp _ _ _
 
 @[simp, reassoc]
-lemma map_lift_comp_preserves_products_iso_hom
-  {J : Type v} (f : J → C) [has_limits C] [has_limits D] (P : C) (g : Π j, P ⟶ f j) :
+lemma map_lift_comp_preserves_products_iso_hom (P : C) (g : Π j, P ⟶ f j) :
   G.map (pi.lift g) ≫ (preserves_products_iso G f).hom = pi.lift (λ j, G.map (g j)) :=
-by { ext, simp [←G.map_comp] }
+by { ext, simp [← G.map_comp] }
 
 end preserve_products
+
+namespace fork
+
+open category_theory.limits.walking_parallel_pair
+
+variables {X Y Z : C} {f g : X ⟶ Y} {h : Z ⟶ X} (w : h ≫ f = h ≫ g)
+
+/--
+The map of a fork is a limit iff the fork consisting of the mapped morphisms
+is a limit.
+This essentially lets us commute `fork.of_ι` with `functor.map_cone`.
+-/
+def equalizer_map_cone_limit :
+  is_limit (G.map_cone (fork.of_ι h w)) ≃
+  is_limit (fork.of_ι (G.map h) (by simp only [←G.map_comp, w]) : fork (G.map f) (G.map g)) :=
+(is_limit.postcompose_hom_equiv (diagram_iso_parallel_pair _) _).symm.trans
+  (is_limit.equiv_iso_limit (fork.ext (iso.refl _) (by simp [fork.ι_eq_app_zero])))
+
+/-- The property of preserving equalizers expressed in terms of forks. -/
+def map_is_limit_of_preserves_of_is_limit [preserves_limit (parallel_pair f g) G]
+  (l : is_limit (fork.of_ι h w)) :
+  is_limit (fork.of_ι (G.map h) (by simp only [←G.map_comp, w]) : fork (G.map f) (G.map g)) :=
+equalizer_map_cone_limit G w (preserves_limit.preserves l)
+
+/-- The property of reflecting equalizers expressed in terms of forks. -/
+def is_limit_of_reflects_of_map_is_limit [reflects_limit (parallel_pair f g) G]
+  (l : is_limit (fork.of_ι (G.map h) (by simp only [←G.map_comp, w]) : fork (G.map f) (G.map g))) :
+  is_limit (fork.of_ι h w) :=
+reflects_limit.reflects ((equalizer_map_cone_limit G w).symm l)
+
+variables (f g)
+
+/--
+If `G` preserves equalizers and `C` has them, then the fork constructed of the mapped morphisms of
+a fork is a limit.
+-/
+def preserves_the_equalizer [has_equalizer f g] [preserves_limit (parallel_pair f g) G] :
```

Comment:
I'm not a huge fan of this name. I think I would prefer something boring like `is_limit_of_has_equalizer_of_preserves_limit`.

Reply:
Works for me, thanks!